src/pages/homepage/hooks/index.tsx

```diff
@@ -2,64 +2,58 @@ import { collection, query, getDocs } from 'firebase/firestore';
 import { db } from '../../../firebase/firebaseSetup';
 import { fullUrl } from '../../../api';
 import { TuserEmail, TMovie } from '../../../types';
-import { useQuery } from 'react-query';
-import { queryKeys } from '../../../react-query/constants';
+import { useState, useEffect } from 'react';
+import { useAuth } from '../../../auth/useAuth';
 
-export const getMovies = async () => {
-  try {
-    const { results: movies } = await (await fetch(fullUrl)).json();
-    return movies;
-  } catch (error) {
-    throw new Error(typeof error);
-  }
+export const getRawMovies = async () => {
+  const { results: movies } = await (await fetch(fullUrl)).json();
+  return movies;
 };
 
-export const useMovies = () => {
-  const fallback: any = [];
-  const {
-    data = fallback,
-    isLoading,
-    isError,
-    error,
-  } = useQuery(queryKeys.movies, getMovies, {
-    refetchOnWindowFocus: false,
+export const getSavedMovies = async (userEmail: TuserEmail) => {
+  const usersMoviesRef = await collection(db, `users/${userEmail}/movies`);
+  const q = query(usersMoviesRef);
+  const querySnapshot = await getDocs(q);
+  let savedMovies: Array<TMovie> = [];
+  querySnapshot.forEach((doc) => {
+    const { id, isAdded, poster_path, release_date, title } = doc.data();
+    savedMovies.push({
+      id,
+      isAdded,
+      poster_path,
+      release_date,
+      title,
+    });
   });
-  return { data, isLoading, isError, error };
+
+  return savedMovies;
 };
 
-export const getUsersSavedMovies = async (userEmail: TuserEmail) => {
-  try {
-    const usersMoviesRef = await collection(db, `users/${userEmail}/movies`);
-    const q = query(usersMoviesRef);
-    const querySnapshot = await getDocs(q);
-    let savedMovies: Array<TMovie> = [];
-    querySnapshot.forEach((doc) => {
-      doc.id &&
-        savedMovies.push({
-          id: doc.data().id,
-          isAdded: doc.data().isAdded,
-          poster_path: doc.data().poster_path,
-          release_date: doc.data().release_date,
-          title: doc.data().title,
-        });
-    });
+export const useFullMovies = () => {
+  const [fullMovies, setFullMovies] = useState<TMovie[]>([]);
+  const [rawMovies, setRawMovies] = useState<TMovie[]>([]);
+  const [savedMovies, setSavedMovies] = useState<TMovie[]>([]);
+  const [error, setError] = useState<string>('');
+  const { user } = useAuth();
+  const userEmail = user?.email;
 
-    return savedMovies;
-  } catch (error) {
-    throw error;
-  }
-};
+  useEffect(() => {
+    getRawMovies()
+      .then((rawMovies) => setRawMovies(rawMovies))
+      .catch((error: string) => setError(error));
+    getSavedMovies(userEmail)
+      .then((savedMovies) => setSavedMovies(savedMovies))
+      .catch((error: string) => setError(error));
+  }, [userEmail]);
+
+  useEffect(() => {
+    if (rawMovies && savedMovies) {
```

Comment:
this shouldnt be in a useeffect but should be in the success part of the first useEffect